scalafmt-core/shared/src/main/scala/org/scalafmt/util/TokenOps.scala

```diff
@@ -59,8 +59,9 @@ object TokenOps {
     }
 
   def forceDocstringBlankLine(token: Token, style: ScalafmtConfig): Boolean = {
-    if (style.optIn.blankLineBeforeDocstring) false
-    else token.is[Comment] && token.syntax.startsWith("/**")
+    style.optIn.forceNewlineBeforeDocstringSummary &&
```

Comment:
why not inline the logic here, as the only place we use it?

```
(style.optIn.forceBlankLineBeforeDocstring ||
  !style.optIn.blankLineBeforeDocstring) &&
  token.is[Comment] && token.syntax.startsWith("/**")
```

Reply:
Because I prefer to encapsulate deprecated flags handling in the `config` package and don't propagate it to the formatting algorithm